sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -614,7 +614,9 @@ void openConnectionsAndInitCaches(Duration aggressiveWarmupDuration) {
     // with a sink and block on the wrapping flux for the specified duration
     private Flux<Void> wrapSourceFluxAndSoftCompleteAfterTimeout(Flux<Void> source, Duration timeout) {
         return Flux.<Void>create(sink -> {
-                    source.subscribe(t -> sink.next(t));
+                    source
+                        .doOnComplete(() -> sink.complete())
```

Comment:
would be doFinally safer here, for case where the source failed

Reply:
Good catch. Fixed in next iteration.